### PR TITLE
Add '--list-checks' flag that lists all implemented checks

### PR DESCRIPTION
**Pull request by vickgoodman**
Usage example:
```
$ beman-tidy --list-checks
[Requirement]    cmake.library_alias
[Recommendation] cmake.library_name
[Recommendation] cmake.project_name
[Recommendation] cpp.extension_identifiers
[Recommendation] cpp.namespace
[Requirement]    directory.docs
[Requirement]    directory.examples
[Requirement]    directory.papers
[Requirement]    directory.sources
[Requirement]    directory.tests
[Recommendation] file.copyright
[Requirement]    file.license_id
[Recommendation] file.names
[Requirement]    file.test_names
[Recommendation] library.name
[Recommendation] license.apache_llvm
[Requirement]    license.approved
[Requirement]    license.criteria
[Requirement]    readme.badges
[Recommendation] readme.implements
[Requirement]    readme.library_status
[Requirement]    readme.license
[Recommendation] readme.purpose
[Recommendation] readme.title
[Requirement]    release.github
[Recommendation] release.godbolt_trunk_version
[Recommendation] release.notes
[Requirement]    repository.code_review_rules
[Requirement]    repository.codeowners
[Requirement]    repository.default_branch
[Recommendation] repository.disallow_git_submodules
[Requirement]    repository.name
[Requirement]    toplevel.cmake
[Requirement]    toplevel.license
[Requirement]    toplevel.readme
```

`$ beman-tidy --list-checks /path/to/repo` outputs the same

### CODE REVIEW

File: beman_tidy/cli.py [hunk 1]
```diff
@@ -41,7 +74,15 @@ def parse_args():
         default=False,
     )
     parser.add_argument(
-        "--checks", help="array of checks to run", type=str, default=None
+        "--list-checks",
+        help="list all implemented checks",
+        action=ListChecksAction,
+    )
```
[the remark this answers is not in the document]
Comment (vickgoodman) [reply beman_tidy/cli.py:84]: I tested this locally and `beman-tidy --list-checks` works without `repo_path`. The custom ListChecksAction exits during argument parsing before argparse performs the final required positional argument validation. Normal commands such as `beman-tidy --verbose` still fail with the expected missing `repo_path` error.

File: README.md [hunk 2]
```diff
@@ -48,19 +48,22 @@ beman-tidy --help
 
 ```shell
 $ beman-tidy --help
-usage: beman-tidy [-h] [--fix-inplace | --no-fix-inplace] [--verbose | --no-verbose] [--require-all | --no-require-all] [--checks CHECKS] [--config CONFIG] repo_path
+usage: beman-tidy [-h] [--version] [--fix-inplace | --no-fix-inplace] [--verbose | --no-verbose] [--require-all | --no-require-all] [--list-checks] [--checks CHECKS] [--config CONFIG]
+                  repo_path
 
 positional arguments:
   repo_path             path to the repository to check
```
[the remark this answers is not in the document]
Comment (vickgoodman) [reply README.md:56]: Not relevant. See prev comment.

File: beman_tidy/cli.py [hunk 3]
```diff
@@ -8,6 +8,39 @@
...
+            dest=dest,
+            nargs=0,
+            **kwargs,
+        )
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        beman_standard_check_config = load_beman_standard_config()
+        if not beman_standard_check_config or len(beman_standard_check_config) == 0:
+            parser.exit(status=1, message="Failed to download the beman standard. STOP.\n")
+
```
[the remark this answers is not in the document]
Comment (vickgoodman) [reply beman_tidy/cli.py:41]: Done.

File: beman_tidy/cli.py [hunk 4]
```diff
@@ -8,6 +8,39 @@
...
 from beman_tidy.lib.utils.git import get_repo_info, load_beman_standard_config
 from beman_tidy.lib.pipeline import run_checks_pipeline
+from beman_tidy.lib.checks.system.registry import get_registered_beman_standard_checks
+
+
+def list_checks(beman_standard_check_config):
+    """
+    Print all implemented beman-tidy checks.
+    """
+    implemented_checks = get_registered_beman_standard_checks()
```
Comment (neatudarius) [inline beman_tidy/cli.py:18]: This works only because the check modules' `@register_...` decorators run as a side effect of the `from ...pipeline import run_checks_pipeline` import above. If that import ever becomes lazy, `--list-checks` silently prints an empty list. Suggest importing the check modules explicitly here (or a comment noting the dependency).

File: beman_tidy/cli.py [hunk 5]
```diff
@@ -8,6 +8,39 @@
...
+        super().__init__(
+            option_strings=option_strings,
+            dest=dest,
+            nargs=0,
+            **kwargs,
+        )
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        beman_standard_check_config = load_beman_standard_config()
+        if not beman_standard_check_config or len(beman_standard_check_config) == 0:
```
Comment (neatudarius) [inline beman_tidy/cli.py:39]: Duplicates the config-load guard in `main()`, and the exit codes differ: here `exit 1`, but `main()`'s early return exits 0. A shared helper would keep them consistent.

**Review by neatudarius — APPROVED** [review]
Works and mirrors the `--version` action pattern. Two inline notes below. Also worth a CLI smoke test for `--list-checks` (asserts exit 0 + non-empty list) — there's no CLI test coverage today.
